### ISSUE TITLE
Update snippets to use modern language features (eg "nameof")

### ISSUE THREAD

**Issue by ljw1004** [dotnet/roslyn#5574]
The "propdp" snippet spits out a dependency property. I think it should spit out the `nameof(.)` operator, rather than spitting out a text literal.

![propdp](https://cloud.githubusercontent.com/assets/3316258/10195413/86e1ea80-675b-11e5-991b-b0feb6c47fc4.png)

**Comment by Pilchie** [comment]
As with #3469, we should probably do a sweep through all snippets.  One consideration from the past is whether we realistically expect customers to use these snippets in scenarios where they would set their "langversion" switch back, since we don't have a mechanism to do different things in snippets.

**Comment by miloush** [comment]
This snippet is actually one of the worst from the very beginning. Apart from `nameof`,
- you have to delete the comment every single time
- it should use `ClassName()` to fill in the ownerclass
- does not support specifying derived metadata
- is cumbersome to use for declaring multiple properties consequently

plus my personal preferences do not improve it,
- I put the static declaration on a single line
- I put static members before instance ones

(and snippets for read-only DP and attached DP are missing completely)
I have submitted Connnect feedback about that long time ago but that seems to be gone now.

As for "langversion", snippets has `<Function>` thing that [I think has great potential but is underdeveloped and] could potentially handle it if needed.

Could perhaps the snippets be put on GitHub or somewhere were people could contribute to them?

**Comment by sharwell** [comment]
Blocked on #31879

**Comment by DaveInCaz** [comment]
The classname issue is mentioned in https://stackoverflow.com/q/35410712/3195477 which also proposes a workaround.

**Comment by CyrusNajmabadi** [comment]
Snippets are being entirely revamped.  So we're unlikely to update these snippets.

**Comment by sharwell** [comment]
@CyrusNajmabadi Can you clarify what additional info is needed?

**Comment by CyrusNajmabadi** [comment]
Need more info on if we're actually intending on giving these snippets a future.

**Comment by sharwell** [comment]
Ah, sounds like the triage was already correct on this issue (blocked on a linked issue).

**Comment by CyrusNajmabadi** [comment]
AFAICT, legacy snippets have no future and are not being maintained at all.  Semantic snippets are the only place we're investing in.  Closing out as out of scope for roslyn now.
